File: backend/models/submission.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Submission struct {
	ID       primitive.ObjectID     `bson:"_id" json:"id,omitempty"`
	Project  string                 `bson:"project" json:"project"`
	ParentID interface{}            `bson:"parentId" json:"parentId"`
	Created  time.Time              `json:"created" bson:"created"`
	Updated  time.Time              `json:"updated" bson:"updated"`
	Title    string                 `json:"title" bson:"title"`
	Status   string                 `json:"status" bson:"status"`
	Author   string                 `json:"author" bson:"author"`
	Group    interface{}            `bson:"group" json:"group"`
	Data     map[string]interface{} `json:"data" bson:"data"`
}

type SubmissionWithChildrenRequest struct {
	Submission Submission   `json:"submission"`
	Children   []Submission `json:"children"`
}

type PartialUpdateRequest struct {
	Submission primitive.ObjectID `bson:"submission" json:"submission"`
	Path       string             `bson:"path" json:"path"`
	Value      interface{}        `bson:"value" json:"value"`
}

File: frontend/src/types/submission.ts
export interface Submission {
  id?: string;
  parentId: string | null;
  viewId: string | null;
  group: string | null;
  project: string;
  created: Date;
  updated: Date;
  title: string;
  status: string;
  author: string;
  data: any;
}

export interface SubmissionWithChildren {
  submission: Submission;
  children: Submission[];
  local: string | null;
}

export interface PAreport {
  id?: string;
  companyCode: string;
  yearMonth: string;
  projectNumber: string;
  projectName: string;
  invoiceNumber: string;
  incomeAccount: string;
  incomeAmountLCSI: string;
  invoiceRecipientName: string;
  invoiceRecipientNumber: string;
  validation: string;
  exSalesValue: string;
  exSalesVODNumber: string;
  exSalesManufacturerNumber: string;
  exSalesManufacturerName: string;
  exSalesBU: string;
  intSalesVendorShare: string;
  intSalesVendorAmount: string;
  intSalesManufacturerNumber: string;
  intSalesManufacturerName: string;
  intSalesBU: string;
}
